src/cargo/util_schemas/manifest.rs

```diff
@@ -1108,27 +1110,130 @@ impl TomlTarget {
     }
 }
 
+macro_rules! str_newtype {
+    ($name:ident) => {
+        #[derive(Serialize, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+        #[serde(transparent)]
+        pub struct $name<T: AsRef<str> = String>(T);
+
+        impl<T: AsRef<str>> $name<T> {
+            pub fn into_inner(self) -> T {
+                self.0
+            }
+        }
+
+        impl<T: AsRef<str>> std::convert::AsRef<str> for $name<T> {
```

Comment:
nit: `std::convert` is in prelude.

```suggestion
        impl<T: AsRef<str>> AsRef<str> for $name<T> {
```